lib/src/components/position_component.dart

```diff
@@ -20,9 +20,9 @@ import 'mixins/hitbox.dart';
 /// rendered automatically when this is updated and rendered.
 /// They are translated by this component's (x,y). They do not need to fit
 /// within this component's (width, height).
-abstract class PositionComponent extends BaseComponent {
+class PositionComponent extends BaseComponent {
```

Comment:
I don't see any reason for making this a concrete class, PositionComponent don't implement a render method and I think it should continue to remain as abstract.

Reply:
Without constructor of PositionComponent, XXComponment cannot initialize PositionComponent.{position|size}.
It's not possible add constructor to abstract class of Dart.

Reply:
I don't think this is true, you totally can add constructors to abstract classes in Dart, I have just tried this small example on dartpad to be sure:

```dart
void main() {
  final b = BlaImplementation(1);
  print(b.x);
}

abstract class Bla {
  int x;
  
  Bla(this.x);
}

class BlaImplementation extends Bla {
  BlaImplementation(int x): super(x);
}
```
I still think that this change should be reverted, I don't see any reason for this class to be concrete.